src/App.tsx

```diff
@@ -0,0 +1,72 @@
+import React, {useEffect, useState} from 'react';
+import {useLocation, useNavigate} from "react-router-dom";
+import RecipeList from "./components/RecipeList";
+import Recipe from "./components/recipe";
+import axios from "axios";
+import RecipeSearchForm from "./components/RecipeSearchForm";
+import {StyledButton} from "./components/styles";
+
+function App() {
+  const [recipes, setRecipes] = useState<Recipe[]>([]);
+  const navigate = useNavigate();
+
+  const retrieveAndShowAllRecipes = () => {
```

Comment:
I think it is a good practice to separate the logic dealing with the server communication from the components. In TK we have a completely separate folder called data containing typing and API methods.

Reply:
I agree. It would be much better to have the logic in a different file. Thank you!